NotPerfect.md

```diff
@@ -0,0 +1,7 @@
+Unless the perfect api interface is "obvious", optimize for fds component creation, rather than getting the api right.
+
+Developers may have the inclination that they cannot contribute new component that they're working on to this new repo. This may be because they're afraid they haven't built a component that accommodates for all future use cases.
```

Comment:
This all makes a lot of good points; to build the most future-proof components there is a tradeoff in the number of components you can build and the number of contributors that we can gain.

There should be a balance here, and I agree that we are currently leaning too heavily on the side of optimizing for the right API rather than for optimizing for component creation.

One big thing we can do to shift the balance is to make the right design principles more "obvious" by documenting conventions and patterns that all reusable components should follow. I will prioritize those docs this week, as I think it could go a long way toward addressing this concern.

It would be great if we could open up FDS to more contributors. The 4.0 release was a good step in this direction, but we can do more.